scripts/tests/test_safe_template_linter.py

```diff
@@ -437,9 +437,12 @@ def test_check_mako_expressions_in_javascript(self, data):
 
         mako_template = textwrap.dedent("""
             <%page expression_filter="h"/>
+            ## switch to JavaScript context
             <script>
                 {expression}
             </script>
+            ## switch back to HTML context
+            ${{x}}
```

Comment:
@nedbat: I noticed the tests were not checking that the context appropriately switched back to an HTML context, so I enhanced these tests.